### PR TITLE
docs: update docs about contribution

### PR DESCRIPTION
**Pull request by tujoworker**

### CODE REVIEW

File: packages/dnb-design-system-portal/src/docs/contribute/deploy.md [hunk 1]
```diff
@@ -8,57 +8,58 @@ icon: 'getting_started'
 
 Publishing new versions to the NPM Package (`@dnb/eufemia`) is handled by a Deploy Server.
 
-## How to create a local package
-
-Run `yarn build:pack` inside `/dnb-eufemia` and you get this file: `/build/dnb-eufemia-v0.0.0-development`.
+## Continuous Integration (CI)
 
-## CI Structure
+The Portal (`dnb-design-system-portal`), all the [icons](https://eufemia.dnb.no/icons/) and the NPM Package (`@dnb/eufemia`) are build. deployed and released by a Continuous Integration (CI) servers.
```
Comment (dinarosv) [inline packages/dnb-design-system-portal/src/docs/contribute/deploy.md:13]:
```suggestion
The Portal (`dnb-design-system-portal`), all the [icons](https://eufemia.dnb.no/icons/) and the NPM Package (`@dnb/eufemia`) are build, deployed and released by a Continuous Integration (CI) server.
```

File: packages/dnb-design-system-portal/src/docs/contribute/deploy.md [hunk 2]
```diff
@@ -8,57 +8,58 @@ icon: 'getting_started'
...
+1. You can merge your _Pull Request_.
+1. A maintainer will create a _Pull Request_ into one of the release branches (`release`,`alpha` or `beta`).
+1. After a release _Pull Request_ got merged, CI Server will deploy the Portal and release a new version to NPM.
 
 ### The Release Branch
 
-Make sure you only make Pull Request from `origin/main` into `origin/release`.
-The Release Branch is more kind of a **secondary branch**. It reflects the latest version, but is actually only used to publish new versions and to make builds for the Portal.
+Make sure you only make _Pull Request_ from `origin/main` into `origin/release`.
+The release branch (`origin/release`) is more kind of a _secondary branch_. It contains the state of the latest version as well as all the git tags – each containing a new version number.
```
Comment (dinarosv) [inline packages/dnb-design-system-portal/src/docs/contribute/deploy.md:29]:
```suggestion
The release branch (`origin/release`) is more like a _secondary branch_. It contains the state of the latest version as well as all the git tags – each containing a new version number.
```

File: packages/dnb-design-system-portal/src/docs/contribute/deploy.md [hunk 3]
```diff
@@ -8,57 +8,58 @@ icon: 'getting_started'
...
-- lib.js
-- index.js
-- package.json
-
-#### Important aspects
-
-- PropTypes are getting wrapped with [this babel plugin](babel-plugin-transform-react-remove-prop-types)) `process.env.NODE_ENV !== "production"`. This way applications in production, will not include `propTypes`. If a component depends to check `propTypes` during runtime, consider to export them, so they not getting removed – or simply use `defaultProps` for the operation.
-- As for now, we use React Class Components, because there may happen a case, where two React instances are used, and that does not work with Hooks. Also, performance is a key factor. But we may consider a rewrite at some point.
\ No newline at end of file
+Run `yarn worspace @dnb/eufemia build:pack` and you should get this file: `/build/dnb-eufemia-v0.0.0-development`.
```
Comment (dinarosv) [inline packages/dnb-design-system-portal/src/docs/contribute/deploy.md:65]:
```suggestion
Run `yarn workspace @dnb/eufemia build:pack` and you should get this file: `/build/dnb-eufemia-v0.0.0-development`.
```

File: packages/dnb-design-system-portal/src/docs/contribute/getting-started.md [hunk 4]
```diff
@@ -66,6 +77,30 @@ Run an environment with either `yarn dev` (for Storybook) or `yarn start` (for E
...
+
+There you can `@import` related **SCSS dependencies**. Like the button component also includes the icon component styles.
+
+#### SCSS Theming
+
+Styles that belong to a "theming footprint" – like colors or individual variants – can be put inside the `/themes` directory:
+
+- `./packages/dnb-eufemia/src/components/button/style/themes/dnb-button-theme-ui.scss`
+
+Theming file names ending with `-ui` will during the package release get packet into the global theming package. More details in the [theming section](/uilib/usage/customisation/theming).
```
Comment (dinarosv) [inline packages/dnb-design-system-portal/src/docs/contribute/getting-started.md:100]:
```suggestion
Theming file names ending with `-ui` will during the package release get packed into the global theming package. More details in the [theming section](/uilib/usage/customisation/theming).
```

File: packages/dnb-design-system-portal/src/docs/contribute/getting-started.md [hunk 5]
```diff
@@ -258,11 +293,73 @@ Make tests for the new component (or for your current issue) and set up screensh
...
+yarn test:screenshots:update breadcrumb
+```
+
+5. How to deal with failing tests on the CI server?
+
+You may check out the logs for detailed reports.
+
+**Failing visual screenshot tests**
+
+If visual screenshot tests are failing on the CI, you can scroll all the we to the end of the logs. There you will find a link **Download the Report** you can use to download a zip file containing the visual test reports and diffs.
```
Comment (dinarosv) [inline packages/dnb-design-system-portal/src/docs/contribute/getting-started.md:355]:
```suggestion
If visual screenshot tests are failing on the CI, you can scroll all the way to the end of the logs. There you will find a link **Download the Report** you can use to download a zip file containing the visual test reports and diffs.
```

**Review by dinarosv — APPROVED** [review]
Thanks for updating this chapter of the contribution docs! 
I was picky in reviewing today, hope it was okay 😄

**Comment by tujoworker** [comment]
Regarding **Table of Contents** links; I added [`yzhang.markdown-all-in-one`](https://marketplace.visualstudio.com/items?itemName=yzhang.markdown-all-in-one) as a recommended VSCode extension. 

It updates the titles/links automatically. But in order to use it, you need to change the language to this one first. 

What do you think?

Screenshot 2022-02-17 at 12 20 23

**Comment by dinarosv** [comment]
> Regarding **Table of Contents** links; I added [`yzhang.markdown-all-in-one`](https://marketplace.visualstudio.com/items?itemName=yzhang.markdown-all-in-one) as a recommended VSCode extension.
> 
> It updates the titles/links automatically. But in order to use it, you need to change the language to this one first.
> 
> What do you think?
> ...

Change the language in VSCode for markdown files you mean? Or something more?

**Comment by tujoworker** [comment]
> Change the language in VSCode for markdown files you mean?

yes

**Pull request merged**